theano/scan_module/scan.py

```diff
@@ -990,6 +992,10 @@ def wrap_into_list(x):
     info['as_while'] = as_while
     info['profile'] = profile
     info['allow_gc'] = allow_gc
+    info['cache_grad'] = cache_grad
+    if cache_grad:
+        warnings.warn('cache_grad works only when Scan Op is associated'
```

Comment:
Is there other condition? I prefer to let it vague, just in case we forget some condition.